src/rules/strictComparisonsRule.ts

```diff
@@ -120,9 +120,68 @@ export class Rule extends Lint.Rules.TypedRule {
     }
 }
 
+function getConstraint(t: ts.Type): ts.TypeReferenceNode | undefined {
+    if (t.getSymbol() === undefined) {
+        return undefined;
+    }
+
+    const paramDeclaration = t.symbol.declarations.find(
+        d => d.kind === ts.SyntaxKind.TypeParameter,
+    ) as ts.TypeParameterDeclaration;
+    if (paramDeclaration === undefined) {
+        return undefined;
+    }
+
+    const constraint = getEffectiveConstraintOfTypeParameter(paramDeclaration);
+    if (constraint === undefined) {
+        return undefined;
+    }
+
+    if (ts.isTypeReferenceNode(constraint)) {
```

Comment:
Ah, the old `ts.is*` trap 😄 you'll want to use `tsutils` for that:

```suggestion
    if (tsutils.isTypeReferenceNode(constraint)) {
```